packages/tfchain_client/src/farms.ts

```diff
@@ -44,4 +68,35 @@ class QueryFarms {
   }
 }
 
-export { QueryFarms, Farm, Certification, FarmingPolicyLimits };
+class Farms extends QueryFarms {
+  constructor(public client: Client) {
+    super(client);
+    this.client = client;
+  }
+
+  @checkConnection
+  async create(options: CreateFarmOptions) {
+    const extrinsic = this.client.api.tx.tfgridModule.createFarm(options.name, options.publicIps);
+    return this.client.patchExtrinsic<Farm>(extrinsic);
+  }
+
+  @checkConnection
+  async addFarmIp(options: AddFarmIPOptions) {
+    const extrinsic = this.client.api.tx.tfgridModule.addFarmIp(options.farmId, options.ip, options.gw);
+    return this.client.patchExtrinsic<Farm>(extrinsic);

```

Comment:
does it return a `Farm` object in adding and removing an ip?

Reply:
Yes. It's weird but yes.
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/ec006417-5658-4202-9edb-a590dd62b750)